nvflare/apis/impl/bcast_manager.py

```diff
@@ -61,6 +69,12 @@ def check_task_exit(self, task: Task) -> Tuple[bool, TaskCompletionStatus]:
             else:
                 clients_responded += 1
 
+        if clients_responded >= num_expected_clients and clients_not_responded == 0:
```

Comment:
can the `clients_responded` be greater than `num_expected_clients`?
If not should we just do `if clients_responded == num_expected_clients`

also, suggest moving this logic after the `# minimum responses received` codes.